listenbrainz/db/model/stats/user/artist.py

```diff
@@ -0,0 +1,39 @@
+import pydantic
+
+from datetime import datetime
+from enum import Enum
+from typing import Optional, List
+
+
+class StatisticsRange(Enum):
+    week = 'week'
+    month = 'month'
+    year = 'year'
+    all_time = 'all_time'
+
+
+class UserArtistRecord(pydantic.BaseModel):
+    artist_msid: Optional[str]
+    artist_mbids: List[str] = []
+    listen_count: int
+    artist_name: str
+
+
+class UserArtistStatForRange(pydantic.BaseModel):
+    to_ts: float
+    from_ts: float
+    count: int
+    artists: List[UserArtistRecord]
+
+
+class UserArtistStatJson(pydantic.BaseModel):
```

Comment:
terminology: I'm not sure about Json - does this imply too much about the underlying structure of the data? I was going to suggest `UserArtistStatResponse` if this was only used to serialise the data when returning it in the API, but I see that it's used in a few other places in the code too, so maybe that term isn't very good.

Reply:
I think Json is the only correct thing that fits here.